### PR TITLE
[Console] Add Metrics Sources for cloudwatch & prometheus

### PR DESCRIPTION
**Pull request by mikaylathompson**
### Description
This task adds Metrics Sources for cloudwatch & prometheus to the console 

### Issues Resolved
[MIGRATIONS-1743](https://opensearch.atlassian.net/browse/MIGRATIONS-1743)

### Testing
Manual testing and unit tests. This will also be used as part of [MIGRATIONS-1699](https://opensearch.atlassian.net/browse/MIGRATIONS-1699), which is an E2E testing task, so that will function as the E2E tests for here.

### Check List
- [x] New functionality includes testing
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+The structure of cli commands is:
+`console [--global-options] OBJECT COMMAND [--options]`
+
+##### Global Options
+The available global options are:
+- `--config-file FILE` to specify the path to a config file (default is `/etc/migration_services.yaml`)
+- `--json` to get output in JSON designed for machine consumption instead of printing to the console
+
+##### Objects
```
Comment (lewijacn) [inline TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/README.md:67]: nit: Could be taxing to keep updated

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/cluster.py [hunk 2]
```diff
@@ -51,21 +44,29 @@ def __init__(self, config: Dict) -> None:
...
 
     def call_api(self, path, method: HttpMethod = HttpMethod.GET) -> requests.Response:
         """
         Calls an API on the cluster.
         """
         if self.auth_type == AuthMethod.BASIC:
-            auth = HTTPBasicAuth(self.auth_details["username"], self.auth_details["password"])
+            assert self.auth_details is not None  # for mypy's sake
+            auth = HTTPBasicAuth(
+                self.auth_details.get("username", None), self.auth_details["password"]
```
Comment (lewijacn) [inline TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/cluster.py:58]: Why use a different pattern for getting username versus getting password here?

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/metrics_source.py [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+    "type": {
+        "type": "string",
+        "allowed": [m.name.lower() for m in MetricsSourceType],
+        "required": True,
+    },
+    "endpoint": {
+        "type": "string",
+        "required": False,
+    },
+    "region": {
```
Comment (lewijacn) [inline TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/metrics_source.py:38]: I assume this is the aws region but others might not

File: TrafficCapture/dockerSolution/src/main/docker/docker-compose.yml
Comment (lewijacn) [inline TrafficCapture/dockerSolution/src/main/docker/docker-compose.yml:1]: [High level comment, disregard placement] Do we configure any logging for the console library currently? Also for our cli we may want to limit stack traces on errors to only print when a say verbose flag is given

**Review by lewijacn — COMMENTED** [review]
Great addition, minor comments added

File: TrafficCapture/dockerSolution/src/main/docker/docker-compose.yml
Comment (mikaylathompson) [reply TrafficCapture/dockerSolution/src/main/docker/docker-compose.yml:1]: Oh, logging is a great point. Yes, we should handle it and agreed there should be a verbose/limited option.

File: TrafficCapture/dockerSolution/src/main/docker/docker-compose.yml [hunk 4]
```diff
@@ -98,9 +98,14 @@ services:
       - ./migrationConsole/lib/console_link/services.yaml:/etc/migration_services.yaml
       # this is a convenience thing for testing -- it should be removed before this makes it to prod.
       - ./migrationConsole/lib/console_link:/root/lib/console_link
+      - ~/.aws:/root/.aws
     environment:
       - MIGRATION_KAFKA_BROKER_ENDPOINTS=kafka:9092
-    #command: ./runTestBenchmarks.sh
+      - AWS_REGION=us-east-2
```
[the remark this answers is not in the document]
Comment (mikaylathompson) [reply TrafficCapture/dockerSolution/src/main/docker/docker-compose.yml:104]: Ah, good point. I put it in to match https://github.com/opensearch-project/opensearch-migrations/blob/433025cc142574a50fdd8a2d3c7c27c67ad5454f/TrafficCapture/dockerSolution/src/main/docker/composeExtensions/otel-everything.yml#L21, which is the otel-to-aws extension for docker compose, so I set this to be able to read those same metrics before I had the `region` option in the yaml. Will remove.

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/README.md (hunk 1 above)
Comment (mikaylathompson) [reply TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/README.md:67]: Fair, but at some point, we will want this to be comprehensive documentation.

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/cluster.py (hunk 2 above)
Comment (mikaylathompson) [reply TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/cluster.py:58]: good question. No good reason as far as I can tell.

**Review by lewijacn — APPROVED** [review]

**Pull request merged**